keps/sig-api-machinery/5589-gogo-dependency/README.md

```diff
@@ -0,0 +1,638 @@
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) within one minor version of promotion to GA
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP describes a path to remove runtime and generation-time dependencies on gogo protobuf,
+while preserving existing behavior and minimizing impact on consumers of Kubernetes API Go types.
+
+## Motivation
+
+The toolchain to generate protobuf marshal / unmarshal support for Kubernetes REST API types
+uses https://github.com/gogo/protobuf. This dependency is unmaintained since 2021.
+
+Reliance on an unmaintained and frozen dependency for complex code generation means
+that addressing any future vulnerability, bug, or feature requirement will require
+switching out that dependency first, possibly in a time-critical scenario.
+
+The gogo-generated code also includes run-time references to gogo packages.
+These package dependencies are propagated to all Kubernetes module consumers,
+and make an unmaintained module appear in their dependency graph.
+Using core Kubernetes libraries should not require consumers to inherit unmaintained dependencies.
+
+The generated code also *appears* to implement interfaces to interoperate with standard Go protobuf libraries
```

Comment:
Do we know off the top of our heads, something that will break if we remove the "interfaces to interoperate with standard Go protobuf libraries" in our ecosystem?